Source/URedis/Private/URedis.cpp

```diff
@@ -24,11 +24,10 @@ void FURedis::Connect(const FStringView host, const int32 port) {
     opts.host = TCHAR_TO_UTF8(host.GetData());
     opts.port = port;
     _instance = MakeUnique<sw::redis::Redis>(opts);
-    check(_instance);
+    check(Ping() == "PONG");
```

Comment:
Maybe, it would be better to make function return boolean instead of void. What do you think?

```cpp
bool Connect(const FStringView host, int32 port = 6379);
```

```cpp
return Ping() == "Pong";
```

Reply:
It's not possible because if the instance is not valid ping will crash, so in any case returning a boolean will be useless.

Reply:
I really don't like concept of 'run until it crashes' ^_____^
I think we should discuss this in more detail.
So, should we leave all error control to the final user? Even in blueprints?

Reply:
There's no control. Think about it. If Redis server is not reachable what's the point of this client? We can log errors/make all methods return booleans, optionals or whatever, but Redis not being available IS a catastrophic failure.

Moreover, in Unreal Engine we don't even have exceptions (by design of the engine itself), and the way of handling catastrophic errors is this one - there's no possible recovery

Reply:
I was talking about using like that:

```cpp
FURedis& URedisModule = FModuleManager::LoadModuleChecked<FURedis>("URedis");

if(URedisModule != nullptr)
{
    try {
        URedisModule.Connect(TEXT("127.0.0.1")):
    }
    catch(...) {
        UE_LOG(LogTemp, Fatal, TEXT("Can't connect to Redis Database"));
    }
}
```

I can't normally imagine program, which runs and then crashes because of the lost connection instead of safely logging and leaving current process (e.g leave to main menu)
Just tell me if it possible to do it out of the plugin. That's what I'm trying to understand.